skbio/sequence/_dna.py

```diff
@@ -123,6 +126,272 @@ def degenerate_map(cls):
     def _motifs(self):
         return _motifs
 
+    def transcribe(self):
+        """Transcribe DNA into RNA.
+
+        DNA sequence is assumed to be the coding strand. Thymine (T) is
+        replaced with uracil (U) in the transcribed sequence.
+
+        Returns
+        -------
+        RNA
+            Transcribed sequence.
+
+        See Also
+        --------
+        translate
+        translate_six_frames
+
+        Notes
+        -----
+        DNA sequence's metadata and positional metadata are included in the
+        transcribed RNA sequence.
+
+        Examples
+        --------
+        Transcribe DNA into RNA:
+
+        >>> from skbio import DNA
+        >>> dna = DNA('TAACGTTA')
+        >>> dna
+        DNA
+        -----------------------------
+        Stats:
+            length: 8
+            has gaps: False
+            has degenerates: False
+            has non-degenerates: True
+            GC-content: 25.00%
+        -----------------------------
+        0 TAACGTTA
+        >>> dna.transcribe()
+        RNA
+        -----------------------------
+        Stats:
+            length: 8
+            has gaps: False
+            has degenerates: False
+            has non-degenerates: True
+            GC-content: 25.00%
+        -----------------------------
+        0 UAACGUUA
+
+        """
+        seq = self._string.replace(b'T', b'U')
+
+        metadata = None
+        if self.has_metadata():
+            metadata = self.metadata
+
+        positional_metadata = None
+        if self.has_positional_metadata():
+            positional_metadata = self.positional_metadata
+
+        # turn off validation because `seq` is guaranteed to be valid
+        return RNA(seq, metadata=metadata,
+                   positional_metadata=positional_metadata, validate=False)
```

Comment:
Good point, forgot that

On Sat, Jul 4, 2015 at 4:52 PM, Jai Ram Rideout <notifications@github.com>
wrote:

> In skbio/sequence/_dna.py
> <https://github.com/biocore/scikit-bio/pull/979#discussion_r33890726>:
>
> > +        0 UAACGUUA
> > +
> > +        """
> > +        seq = self._string.replace(b'T', b'U')
> > +
> > +        metadata = None
> > +        if self.has_metadata():
> > +            metadata = self.metadata
> > +
> > +        positional_metadata = None
> > +        if self.has_positional_metadata():
> > +            positional_metadata = self.positional_metadata
> > +
> > +        # turn off validation because `seq` is guaranteed to be valid
> > +        return RNA(seq, metadata=metadata,
> > +                   positional_metadata=positional_metadata, validate=False)
>
> lowercase defaults to False so I don't think setting it will skip
> anything.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/979/files#r33890726>.
>